android/src/main/java/com/imagepicker/ImagePickerModule.java

```diff
@@ -127,21 +127,27 @@ public void launchImageLibrary(final ReadableMap options, final Callback callbac
         Intent libraryIntent;
         requestCode = REQUEST_LAUNCH_LIBRARY;
 
-        if (this.options.mediaType.equals(mediaTypeVideo)) {
-            libraryIntent = new Intent(Intent.ACTION_PICK);
-            libraryIntent.setType("video/*");
-        } else if (this.options.mediaType.equals(mediaTypePhoto)) {
+        boolean isSingleSelect = this.options.selectionLimit == 1;
+        boolean isPhoto = this.options.mediaType.equals(mediaTypePhoto);
+        boolean isVideo = this.options.mediaType.equals(mediaTypeVideo);
+
+        if(isSingleSelect && (isPhoto || isVideo)) {
             libraryIntent = new Intent(Intent.ACTION_PICK);
-            libraryIntent.setType("image/*");
         } else {
             libraryIntent = new Intent(Intent.ACTION_GET_CONTENT);
-            libraryIntent.setType("*/*");
-            libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*", "video/*"});
             libraryIntent.addCategory(Intent.CATEGORY_OPENABLE);
+            libraryIntent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
         }
 
-        if (this.options.selectionLimit != 1) {
-            libraryIntent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
+        if(isPhoto) {
+            libraryIntent.setType("image/*");
+            libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*"});
```

Comment:
`libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*"}); is not required when u setType. The Mime type we use in setType will be used.